aspnetcore/security/authentication/google-logins.md

```diff
@@ -0,0 +1,142 @@
+
+![image](sociallogins/_static/GoogleConsoleLogin.png)
+
+* You will be redirected to API Manager Library page:
+
+![image](sociallogins/_static/GoogleConsoleSwitchboard.png)
+
+* Tap **Create a project** and enter your application name:
+
+![image](sociallogins/_static/GoogleConsoleNewProj.png)
+
+* You will be redirected back to the Library page allowing you to choose features your new app will support. Find *Google+ API* in the list and click on its link to add the API feature:
+
+![image](sociallogins/_static/GoogleConsoleChooseApi.png)
+
+* The page for the newly added API will be displayed. Tap **Enable** to add Google+ login feature to your app:
+
+![image](sociallogins/_static/GoogleConsoleEnableApi.png)
+
+* After enabling the API, tap **Go to Credentials** to configure the secrets:
+
+![image](sociallogins/_static/GoogleConsoleGoCredentials.png)
+
+* Choose:
+   * **Google+ API**
+   * **Web server (e.g. node.js, Tomcat)**, and
+   * **User data**:
+
+![image](sociallogins/_static/GoogleConsoleChooseCred.png)
+
+* Tap **What credentials do I need?** which will take you to the second step of app configuration:
+
+![image](sociallogins/_static/GoogleConsoleCreateClient.png)
+
+* Because we are creating a Google+ project with just one feature (logins), we can enter the same **Name** for the OAuth 2.0 client ID as the one we used for the project.
+
+* Enter your current site URL with *signin-google* appended into the **Authorized redirect URIs** field. For example, https://localhost:44320/**signin-google**.
+
+> [!NOTE]
+> You don't need to configure **signin-google** as a route in your app. The ASP.NET Core team's implementation of the OAuth flow will create a temporary socket (called a *backchannel*) that listens at this route just for the duration of the OAuth flow.
+
+* Tap **Create client ID**, which will take you to the third step:
+
+![image](sociallogins/_static/GoogleConsoleAddCred.png)
+
+* Enter your public facing *Email address* and the *Product name* shown for your app when Google+ prompts the user to login.
+
+* Tap **Continue** to proceed to the last step:
+
+![image](sociallogins/_static/GoogleConsoleFinish.png)
+
+* Tap **Download** to save a JSON file with application secrets, and **Done** to complete creation of the new app.
+
+## Storing Google ClientID and ClientSecret
+
+Link sensitive settings like Google *ClientID* and *ClientSecret* to your application configuration by using the [Secret Manager tool](../app-secrets.md) instead of storing them in your configuration file directly, as described in the [previous section](sociallogins.md).
```

Comment:
It's a bit awkward because sometimes it refers to values on provider's config page which may have spaces in them (for example App Secret) and sometimes to config setting name in app code (for example AppSecret). I highlighted both with code blocks for now.